packages/meditrak-server/src/tests/routes/GETAnswers.test.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { expect } from 'chai';
+import {
+  buildAndInsertSurveys,
+  buildAndInsertSurveyResponses,
+  findOrCreateDummyRecord,
+  generateTestId,
+} from '@tupaia/database';
+import { Authenticator } from '@tupaia/auth';
+import { TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, BES_ADMIN_PERMISSION_GROUP } from '../../permissions';
+import { TestableApp } from '../TestableApp';
+import { prepareStubAndAuthenticate } from './utilities/prepareStubAndAuthenticate';
+
+describe('Permissions checker for GETAnswers', async () => {
+  const DEFAULT_POLICY = {
+    DL: ['Public'],
+    KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
+    SB: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Royal Australasian College of Surgeons'],
+    VU: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
+    LA: ['Admin'],
+  };
+
+  const BES_ADMIN_POLICY = {
+    SB: [BES_ADMIN_PERMISSION_GROUP],
+  };
+
+  const app = new TestableApp();
+  const { models } = app;
+  let vanuatuAdminResponseId;
+  let vanuatuDonorResponseId;
+  let vanuatuAdminAnswers;
+  let vanuatuDonorAnswers;
+  let filterString;
+
+  before(async () => {
+    const adminPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Admin',
+    });
+    const donorPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Donor',
+    });
+
+    const vanuatuCountry = await findOrCreateDummyRecord(models.country, { code: 'VU' });
+
+    const vanuatuEntity = await findOrCreateDummyRecord(models.entity, {
+      country_code: vanuatuCountry.code,
+    });
+
+    await buildAndInsertSurveys(models, [
+      {
+        code: 'TEST_SURVEY_1',
+        name: 'Test Survey 1',
+        permission_group_id: adminPermissionGroup.id,
+        questions: [{ code: 'TEST_SURVEY_11', text: 'Best spice' }],
+      },
+      {
+        code: 'TEST_SURVEY_2',
+        name: 'Test Survey 2',
+        permission_group_id: donorPermissionGroup.id,
+        questions: [{ code: 'TEST_SURVEY_21', text: 'Worst spice' }],
+      },
+    ]);
+
+    // It's much easier to keep track of the different ids this way
+    // rather than trying to extract the correct ids from the models array
+    vanuatuAdminResponseId = generateTestId();
+    vanuatuDonorResponseId = generateTestId();
+
+    const surveyResponseModels = await buildAndInsertSurveyResponses(models, [
```

Comment:
Can we just call these `surveyResponses`? What is returned by that function is an array of instances of the `SurveyResponseType` class, rather than the `SurveyResponseModel` class